iotdb/src/main/java/org/apache/iotdb/db/query/reader/mem/MemChunkReader.java

```diff
@@ -65,27 +77,45 @@ public TimeValuePair next() {
   }
 
   @Override
-  public void skipCurrentTimeValuePair() {
-    next();
+  public TimeValuePair current() throws IOException {
+    if (!hasCachedTimeValuePair) {
+      cachedTimeValuePair = timeValuePairIterator.next();
+      hasCachedTimeValuePair = true;
+    }
+    return cachedTimeValuePair;
   }
 
   @Override
-  public void close() {
-    // Do nothing because mem chunk reader will not open files
+  public BatchData nextBatch() throws IOException {
+    BatchData batchData = new BatchData(dataType, true);
+    if (hasCachedTimeValuePair) {
+      hasCachedTimeValuePair = false;
+      batchData.putTime(cachedTimeValuePair.getTimestamp());
+      batchData.putAnObject(cachedTimeValuePair.getValue().getValue());
+    }
+    while (timeValuePairIterator.hasNext()) {
+      TimeValuePair timeValuePair = timeValuePairIterator.next();
+      if (filter == null || filter
+          .satisfy(timeValuePair.getTimestamp(), timeValuePair.getValue().getValue())) {
+        batchData.putTime(timeValuePair.getTimestamp());
+        batchData.putAnObject(timeValuePair.getValue().getValue());
+      }
+    }
+    return batchData;
   }
 
   @Override
-  public boolean hasNextBatch() {
-    return false;
+  public void close() {
+    // Do nothing because mem chunk reader will not open files
   }
 
   @Override
-  public BatchData nextBatch() {
+  public PageHeader nextPageHeader() throws IOException {
     return null;
   }
 
   @Override
-  public BatchData currentBatch() {
-    return null;
+  public void skipPageData() throws IOException {
+    nextBatch();
```

Comment:
Because there is no page in memory, all the data in memory is treated as a page. So skipPageData() is equivalent to skip all data in memory. But in fact, this method is not called. Do you think it's necessary to make this method an empty method or throw an exception when calling it?

Reply:
Better make it throw an exception in case that someone misuses this method.